corehq/apps/users/tests/test_email_events.py

```diff
@@ -36,3 +42,51 @@ def test_bounced_event(self):
         invite = Invitation.objects.get(uuid=self.invite.uuid)
 
         self.assertEqual(invite.email_status, InvitationStatus.BOUNCED)
+
+
+class TestUserLimitWarnings(TestSnsEmailBase):
+    def setUp(self):
+        self.domain = create_domain("testapp")
+        initialize_domain_with_default_roles(self.domain.name)
+        self.recipients = ["orange@gmail.com", "purple@gmail.com"]
+        self.domain_admin = WebUser.create(self.domain.name, self.recipients[0], "123", None, None, is_admin=True)
+        self.domain_admin.save()
+        self.billing_admin_user = WebUser.create(self.domain.name, self.recipients[1], "123", None, None)
+        role = UserRole.objects.get(domain=self.domain, name=UserRolePresets.BILLING_ADMIN)
+        self.billing_admin_user.set_role(self.domain.name, role.get_qualified_id())
+        self.billing_admin_user.save()
+
+    def tearDown(self):
+        self.domain_admin.delete(self.domain.name, None)
+        self.billing_admin_user.delete(self.domain.name, None)
+        self.domain.delete()
+
+    def test_limit_emails(self):
+        plan_limit = 100
+
+        # email should not send if user count is still below the 90% threshold
+        prev_user_count = 75
+        post_user_count = 85
```

Comment:
Because off-by-one errors are so common, I generally like to test right on the boundaries. Since its greater-than-or-equal comparisons, that would mean no emails up to 89 users, and ensuring an email is sent on exactly 90 users.